### PR TITLE
Auto fix common single words

### PR DESCRIPTION
**Pull request by m-nash**
# Description
In mgmt plane there are tons of copies of the same class that use a single word name.  The main common ones are Sku, Plan, and Usage.  This PR prepends those classes with the library name to make them unique.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/TestProjects/MgmtRenameRules/Generated/Models/MgmtRenameRulesPlan.cs [hunk 1]
```diff
@@ -8,19 +8,19 @@
 namespace MgmtRenameRules.Models
 {
     /// <summary> Specifies information about the marketplace image used to create the virtual machine. This element is only used for marketplace images. Before you can use a marketplace image from an API, you must enable the image for programmatic use.  In the Azure portal, find the marketplace image that you want to use and then click **Want to deploy programmatically, Get Started -&gt;**. Enter any required information and then click **Save**. </summary>
-    public partial class Plan
+    public partial class MgmtRenameRulesPlan
```
Comment (ShivangiReja) [inline test/TestProjects/MgmtRenameRules/Generated/Models/MgmtRenameRulesPlan.cs:11]: Can we add the test cases to verify the names?

File: src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        };
+
+        public static void Update(IEnumerable<Schema> allSchemas)
+        {
+            foreach (var schema in allSchemas)
+            {
+                if (_schemasToChange.Contains(schema.Name))
+                {
+                    string suffix = schema.Language.Default.Name.Equals("Resource") ? "Data" : string.Empty;
+                    schema.Language.Default.Name = MgmtContext.Context.DefaultNamespace.Split('.').Last() + schema.Name + suffix;
```
Comment (fengzhou-msft) [inline src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs:30]: Does this also apply to Azure.ResourceManager, then are we OK with names like `ResourceManagerSku`?

File: src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{
+    internal static class CommonSingleWordModels
+    {
+        private static readonly HashSet<string> _schemasToChange = new HashSet<string>()
+        {
+            "Sku",
+            "Plan",
+            "Usage",
+            "Resource"
+        };
```
Comment (ArthurMa1978) [inline src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs:21]: Ok to have static now, but consider to move out into a config

File: src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs (hunk 2 above)
Comment (ArthurMa1978) [reply src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs:30]: same question, and I don't like the name `ResourceManagerSku`

Comment (m-nash) [reply src/AutoRest.CSharp/Mgmt/Decorator/CommonSingleWordModels.cs:30]: Changing to Arm when its Azure.ResourceManager namespace

**Review by ShivangiReja — APPROVED** [review]
LGTM

**Pull request merged**